daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/DPath.scala

```diff
@@ -159,12 +167,12 @@ final class RuntimeExpressionDPath[T <: AnyRef](qn: NamedQName, tt: NodeInfo.Kin
    * so as to detect deadlocks.
    */
   def evaluateForwardReferencing(state: ParseOrUnparseState, whereBlockedInfo: Suspension): Maybe[T] = {
-    var value: Maybe[AnyRef] = Nope
+    var value:Maybe[T] = Nope
```

Comment:
This type of change occurs in several places. Due to type erasure, it does not actually do much of anything.

I assume the previous use of Maybe[AnyRef] was just an oversight.

I was not systemetic in looking for these. Where you see me fix this, what probably happened was I accidently changed the type to DataValue, then set it back to Maybe[T] when I realized my mistake. Fixing it is not acutally relevent to what I was trying to do, but I see no reason to go out of my way to avoid fixing it in the places where I did.